### PR TITLE
added a sparse-compatible ID column prefixing

### PR DESCRIPTION
**Pull request by retdop**
As discussed in #162, here is how I made keppler-mapper sparse-compatible with my use case.

### CODE REVIEW

File: kmapper/kmapper.py [hunk 1]
```diff
@@ -498,7 +499,10 @@ def map(
         # Prefix'ing the data with an ID column
         ids = np.array([x for x in range(lens.shape[0])])
         lens = np.c_[ids, lens]
-        X = np.c_[ids, X]
+        if type(X) == csr.csr_matrix:
```
Comment (sauln) [inline kmapper/kmapper.py]: Could you please add an example in the docstring above and mention in the Parameters section that `csr_matrix` is a valid format?

File: kmapper/kmapper.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
     graph_data_distribution,
     colorscale_default,
 )
+from scipy.sparse import hstack, csr
```
Comment (sauln) [inline kmapper/kmapper.py]: Could you please combine this import with the other `scipy.sparse` imports. It should read
```
from scipy.sparse import issparse, hstack, csr
```

**Comment by sauln** [comment]
Thanks for submitting this! This is a nice addition to the library. There are just a couple little things to add that I mentioned in the review. Once those are done, I'll be glad to merge the PR.

Thanks!

**Comment by retdop** [comment]
I committed with the requested changes.  What do you think?

**Comment by sauln** [comment]
Looks good! tests are passing and code coverage has actually gone up.

Thanks for this contribution!

**Pull request merged**